src/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -165,6 +165,13 @@ public static int Run(string[] args)
                     msbuildArgs.Add(argRoot.Value);
                 }
 
+                // Get runsetings options specified after -- 
+                if (cmd.RemainingArguments != null && cmd.RemainingArguments.Count > 0)
+                {
+                    var runSettingsOptions = GetRunSettingsOptions(cmd.RemainingArguments);
+                    msbuildArgs.Add(string.Format("/p:VSTestCLIRunSettings=\"{0}\"", string.Join(";", runSettingsOptions)));
```

Comment:
GetSemiColonEscapeArgs is just replacing semi colon with %3b in all the elements.
GetRunSettingsOptions is extracting the run settings after '--' and replacing semi colon with %3b as well.